src/plugins/intel_cpu/src/nodes/kernels/scaled_attn/executor_pa.cpp

```diff
@@ -0,0 +1,1595 @@
+                    for (size_t h = hk * _h_each_group_len; h < (hk + 1) * _h_each_group_len; h++) {
+                        attn_acc_value_block(_output_bhl.ptr<float>(ithr, b, pq, h),
+                                             _weight_bhl.ptr<float>(b, h, pq) + pv,
+                                             v,
+                                             _S,
+                                             std::min(_block_size, context_len - pv));
+                    }
+                }
+            }
+        });
+
+        parallel_for3d(B, _H, q_len, [&](size_t b, size_t h, size_t pq) {
+            auto* temp = _output_bhl.ptr<float>(0, b, pq, h);
+            size_t temp_stride = _output_bhl.stride(0);
+            auto* dst = output_emb.ptr<DATA_TYPE>(b, pq, h * _S);
+            attn_reduce(dst, temp, _nthr, _S, temp_stride);
+        });
+    }
+};
+
+template <typename DATA_TYPE, typename KVCACHE_TYPE>
+struct MHAMultiple {
+    MHAHelper<DATA_TYPE, KVCACHE_TYPE>& _helper;
+
+    MHAMultiple(MHAHelper<DATA_TYPE, KVCACHE_TYPE>& helper) : _helper(helper) {}
+
+    void operator()(PlainTensor& query,
+                    PlainTensor& present_key,
+                    PlainTensor& present_value,
+                    PlainTensor& output_emb,
+                    const PlainTensor& block_tables,
+                    size_t max_context_len,
+                    const PlainTensor& context_lens) {
+        auto B = query.m_dims[0];
+        auto Hk = present_value.m_dims[1];
+        constexpr bool q_is_bf16 = precision_of<DATA_TYPE>::value == ov::element::bf16;
+        constexpr bool q_cache_is_same = precision_of<DATA_TYPE>::value == precision_of<KVCACHE_TYPE>::value;
+
+        // buffer for transpose and repack
+        _helper.init_reorder_buffers(B, block_tables.m_dims[1]);
+
+        // packed k, v
+        parallel_for3d_dynamic(B, block_tables.m_dims[1], Hk, [&](size_t b, size_t kv_block, size_t hk) {
+            auto block_number = block_tables.ptr<int32_t>(b)[kv_block];
+            if (block_number < 0)
+                return;
+            auto ithr = parallel_get_thread_num();
+            auto* k_ptr = present_key.ptr<KVCACHE_TYPE>(block_number, hk);
+            auto* v_ptr = present_value.ptr<KVCACHE_TYPE>(block_number, hk);
+            transpose_16Nx16K(_helper._qk_scratch_b.template ptr<DATA_TYPE>(b, kv_block, hk),
+                k_ptr,
+                _helper._output.template ptr<DATA_TYPE>(ithr),
+                _helper._block_size,
+                _helper._S, _helper._block_size, _helper._S);
+            if (q_is_bf16) {
+                pack_32Nx32K(_helper._wv_scratch_b.template ptr<DATA_TYPE>(b, kv_block, hk),
+                    v_ptr,
+                    _helper._output.template ptr<DATA_TYPE>(ithr),
+                    _helper._block_size,
+                    _helper._S,
+                    _helper._S);
+            } else {
+                if (!q_cache_is_same) {
```

Comment:
Maybe you should clarify the `else` condition. The `else` branch means query is not bf16 and the query's precision is not same with KV_cache. When will this happen, when the query is `fp32` while KV cache is stored in `U8` ?

Reply:
Yes, comments added.